src/cargo/util/lints.rs

```diff
@@ -880,6 +882,111 @@ pub fn unused_dependencies(
     Ok(())
 }
 
+pub fn unexpected_target_cfgs(
+    ws: &Workspace<'_>,
+    pkg: &Package,
+    path: &Path,
+    error_count: &mut usize,
+    gctx: &GlobalContext,
+) -> CargoResult<()> {
+    fn warn_on_unexpected_cfgs(
+        gctx: &GlobalContext,
+        check_cfg: &CheckCfg,
+        cfg_expr: &CfgExpr,
+        lint_level: LintLevel,
+        error_count: &mut usize,
+        path: Option<&Path>,
+        suffix: &str,
+    ) -> CargoResult<()> {
+        let prefix = if let Some(path) = path {
+            format!("{path}: ", path = path.display())
+        } else {
+            "".to_string()
+        };
+
+        let mut emit_lint = |msg| match lint_level {
+            LintLevel::Warn => gctx.shell().warn(msg),
+            LintLevel::Deny | LintLevel::Forbid => {
+                *error_count += 1;
+                gctx.shell().error(msg)
+            }
+            LintLevel::Allow => Ok(()),
+        };
+
+        cfg_expr.try_for_each(|cfg| -> CargoResult<()> {
+            let (name, value) = match cfg {
+                Cfg::Name(name) => (name, None),
+                Cfg::KeyPair(name, value) => (name, Some(value.to_string())),
+            };
+
+            match check_cfg.expecteds.get(name) {
+                Some(ExpectedValues::Some(values)) if !values.contains(&value) => {
+                    let value = if let Some(ref value) = value {
+                        value
+                    } else {
+                        "(none)"
+                    };
+                    emit_lint(format!(
+                        "{prefix}unexpected `cfg` condition value: `{value}` for `{cfg}` in `[target.'cfg({cfg_expr})'{suffix}]`"
```

Comment:
Not seeing a `(none)` in a test case but in my looking at this string, it feels weird to have a
```
some_path/Cargo.toml: unexpected `cfg` condition value: `(none)` for `unix` in `[target.'cfg(unix="foo"`.dependencies]`
```
specifically having a description in quotes that implies a literal

Reply:
(or whatever `(none)` maps to)